src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -941,6 +941,16 @@ private function step_in_body() {
 				$this->reconstruct_active_formatting_elements();
 				$this->insert_html_element( $this->state->current_token );
 				return true;
+
+			/*
+			 * > A start tag whose tag name is one of: "param", "source", "track"
+			 */
+			case '+PARAM':
+			case '+SOURCE':
+			case '+TRACK':
+				$this->insert_html_element( $this->state->current_token );
+				$this->state->stack_of_open_elements->pop();
```

Comment:
we shouldn't be popping these off here otherwise we won't be able to visit the nodes. we should have a failing test at this point, so maybe our tests are lacking.

`step()` handles popping void and self-closing elements _after_ we've had the chance to inspect them. we need to add `PARAM` to the list of void elements. as with `KEYGEN`, it's not listed in the spec because it's a deprecated element, but it's still void.

Reply:
There was a failing test for this but it was on another branch. That's been merged and I've rebased this so the test did discover the issue. I've removed the immediate pop and with the `is_void` check this is now working correctly and the test is passing.